### PR TITLE
Add price to availability post and booking responses

### PR DESCRIPTION
**Pull request by dhedlund**
A subset of pricing is important to re-add to the core spec, as there is no way to estimate how much money a booking may cost to the reseller.

* By the time the POST availability call is made, the server representing the supplier should be able to give an estimated price for reserving the booking.  This price is not guaranteed to be the same by the time a booking reservation call is made, but should try to reflect the price that would've been locked in had the booking reservation taken place at that time.

* When a booking is reserved, the server representing the supplier should be able to lock in a price that represents the amount owned to the supplier if the booking is confirmed.  A reseller may choose to charge a different price to a customer.

_NOTE:_ Descriptions were added to price fields, but it looks like SwaggerHub ignores these if there is a `$ref` defined.

### CODE REVIEW

File: docs/openapi_v1.0.0-alpha.yaml [hunk 1]
```diff
@@ -188,18 +188,24 @@ paths:
...
                 items:
-                  $ref: '#/components/schemas/AvailabilityStatus'
+                  $ref: '#/components/schemas/AvailabilityStatusWithPrice'
                 example:
                   - id: '28271273-a317-40fc-8f42-79725a7072a3'
                     localDateTimeStart: '2019-10-31T08:30:00Z'
                     localDateTimeEnd: '2019-10-31T10:00:00Z'
                     status: 'AVAILABLE'
                     vacancies: 100
+                    price:
```
Comment (ollym) [inline docs/openapi_v1.0.0-alpha.yaml]: Can we rename "price" to "total" as the word price indicates its something per-ticket for example

File: docs/openapi_v1.0.0-alpha.yaml [hunk 2]
```diff
@@ -591,6 +610,10 @@ components:
           description: |
             A valid option ID that matches the `id` returned from `GET /suppliers/{supplierId}/products`.
           example: 'LR1-01'
+        price:
```
Comment (ollym) [inline docs/openapi_v1.0.0-alpha.yaml]: Same here

File: docs/openapi_v1.0.0-alpha.yaml (hunk 1 above)
Comment (gimpster) [reply docs/openapi_v1.0.0-alpha.yaml]: Agreed. I think that would be more clear.

Comment (dhedlund) [reply docs/openapi_v1.0.0-alpha.yaml]: I've changed the price field to total.

File: docs/openapi_v1.0.0-alpha.yaml (hunk 2 above)
Comment (dhedlund) [reply docs/openapi_v1.0.0-alpha.yaml]: Changed.

**Comment by dhedlund** [comment]
> I also feel we need to have some ability to show per-unit pricing?

I feel like per-unit pricing could be proposed as a separate PR and feedback cycle; the goal of this PR is to be able to establish the potential liability of a reseller to the supplier as a result of confirming a booking with units requested.

Some things to consider on a per-unit pricing:  The responses of the availability endpoints don't break anything out by units in the response, not even quantity; maybe the POST /availability response needs to be reworked a bit.  There is also the issue that the total could be affected by dynamic pricing rules, there could be additional non-unit costs rolled up into the total, or even the question of whether the per-unit price would be expected to include taxes and fees.

**Comment by ollym** [comment]
Ah ok that wasn't clear then either. I interpreted this as the retail cost, not the net/wholesale/buy rate.
Two very different things. Can we call it "net" then instead of "total" if that's what you intended it for?

**Comment by chrisonhismac** [comment]
We specifically left pricing out of core spec as it was determined it added too many variables and it isnt a core data requirement to secure a booking. You do not need pricing to do a booking which is what the core spec is for. 

Pricing means many things to different partners and doesn't allow for the nuance of bundled items etc. (Reminder that i was the one who initially fought for it to be included in the meetings but was convinced that it shouldn't be in)

I think it should be the first capability `octo-capability-pricing` which would mean it would come back with requests anyway should both parties allow for pricing (which not everyone does)

**Comment by chrisonhismac** [comment]
> I also feel we need to have some ability to show per-unit pricing?

You're describing a capability

**Comment by gimpster** [comment]
@dhedlund @chrisonhismac @ollym Let's finalize the decision about whether to move forward with this change during the sync today.

**Comment by anush** [comment]
I'm in favor of this change. 

Not knowing what net amount is receivable by the supplier makes it impossible for us to display the reservation -- I believe every OTA we're integrated with today provides this data. While we can _assume_ what the amount _should_ be based on their pricing configuration in our system, but there's no guarantee this is accurate. We'd rather not show incorrect information, especially when it comes to monetary values. We don't have the concept of "unknown" amount that we can show for a reservation. So from our perspective, this is a real core requirement, not a capability.

File: docs/openapi_v1.0.0-alpha.yaml [hunk 3]
```diff
@@ -440,6 +458,7 @@ components:
       required:
         - uuid
         - status
+        - total
```
Comment (anush) [inline docs/openapi_v1.0.0-alpha.yaml]: Are we changing this to `net`?

File: docs/openapi_v1.0.0-alpha.yaml [hunk 4]
```diff
@@ -188,18 +188,24 @@ paths:
               schema:
                 type: array
                 items:
-                  $ref: '#/components/schemas/AvailabilityStatus'
+                  $ref: '#/components/schemas/AvailabilityStatusWithPrice'
```
Comment (anush) [inline docs/openapi_v1.0.0-alpha.yaml]: I'm with @chrisonhismac wrt including price in availability. That should be a capability and future core. But what I do agree with is that the `net` should be part of the booking request.

Comment (chrisonhismac) [reply docs/openapi_v1.0.0-alpha.yaml]: Net to who? The operator? The originator of the booking? OTA? Can we assume that Net is the same to all parties? There is too much variety in pricing to have it in core.

Comment (anush) [reply docs/openapi_v1.0.0-alpha.yaml]: Net payable to the supplier. If there is a daisy chain, the net may be different across each part of the chain, but the definition remains the same -- its the net amount payable by the caller to the callee.

Comment (chrisonhismac) [reply docs/openapi_v1.0.0-alpha.yaml]: I don't see this as a required field but I've had my say. The name should be supplier_net if that's the intent. Net isn't descriptive enough.

**Comment by dhedlund** [comment]
Closing for now, will address as a capability.

**Pull request closed without merging**